micro-rdk/src/common/registry.rs

```diff
@@ -595,12 +615,12 @@ mod tests {
 
         // register fake board
         common::board::register_models(&mut registry);
-        let ctor = registry.get_board_constructor("fake".to_string());
+        let ctor = registry.get_board_constructor("fake");
         assert!(ctor.is_ok());
 
         // register test sensor
         assert!(registry
-            .register_sensor("test_sensor", &TestSensor::from_config)
+            .register_sensor("test_sensor".to_string(), &TestSensor::from_config)
```

Comment:
It looks to me like the `to_string`s here and below on `register_x` calls are no longer needed.